### DISCUSSION TITLE
classes/class_directionallight2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#247] [User-contributed notes]
# classes/class_directionallight2d

Inherits: Light2D< Node2D< CanvasItem< Node< Object Directional 2D light from a distance. Description: A directional light is a type of Light2D node that models an infinite number of parallel rays ...

https://docs.godotengine.org/en/stable/classes/class_directionallight2d.html

**Comment by tot121212** [comment]
To all future people trying to use the DirectionalLight2D,
Be aware of this, as of 11/10/24 - https://github.com/godotengine/godot/issues/64939 , https://github.com/godotengine/godot/pull/93881
The Z Range does not currently work well, so shadows will forever be drawn on top of sprites when not wanted.
Unfortunately we will have to wait until the commit is merged.
Bummer, as I want cool shadows....
